tests/drivers/build_all/display/app.overlay

```diff
@@ -161,6 +161,28 @@
 			};
 		};
 
+		test_mipi_dbi_xfr_16bit_write_only {
+			compatible = "zephyr,mipi-dbi-spi";
+			status = "okay";
+			dc-gpios = <&test_gpio 0 0>;
+			spi-dev = <&test_spi>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			xfr-min-bits = <MIPI_DBI_SPI_XFR_16BIT>;
+			write-only;
```

Comment:
Minor nit- if we drop the `write-only` here, we would get better CI coverage, right?

Reply:
Well, basically the `write-only` property is irrelevant here for the resulting test binary, as the entire part of the read functions will be compiled in anyway by the `test_mipi_dbi_ili9342c` node above (without `write-only`). However, since in the context of the now new 16-bit support it is never possible to read, the explicit setting of `write-only` would not only be correct here at device tree level, but more necessary. This means that there is a notation right here in the test that should also be used in shields or boards for 16-bit accesses in order to avoid unnecessary code in the driver. The entire `MIPI_DBI_SPI_*_REQUIRED` macros already provide the correct opt-out strategy. The CI coverage should therefore not be disturbed or changed by this.

Reply:
Yeah, makes sense- I was thinking that the 16 bit support might disable reads based on the code here. Maybe it is worth documenting that 16 bit support and read support cannot coexist?

Reply:
Currently we (I) know only about the two rare types of displays `[referred in the beginning of this PR](https://github.com/zephyrproject-rtos/zephyr/pull/67089#issue-2061591661)`. Even this hardware never allows write only. That's why I add exactly that real existent case as additional test device. But that doesn't mean that everytime we see a display with 16 bit access also support read access. So I never document that explicit exclusion pn driver level. That exist only on real device level.